packages/frontend/src/routes/app/posts/[id]/page.tsx

```diff
@@ -77,6 +85,49 @@ const PostDetailsPage: React.FC = () => {
         setIsOpenCommnet((prev) => !prev)
     }
 
+    const handleVote = async (voteType: VoteAction): Promise<boolean> => {
+        if (!post) return false
+        try {
+            if (post.isMine) {
+                let cancelAction: VoteAction
+                if (post.finalAction === VoteAction.UPVOTE) {
+                    cancelAction = VoteAction.CANCEL_UPVOTE
+                    post.upCount -= 1
+                } else if (post.finalAction === VoteAction.DOWNVOTE) {
+                    cancelAction = VoteAction.CANCEL_DOWNVOTE
+                    post.downCount -= 1
+                } else {
+                    throw new Error('Invalid finalAction')
+                }
+
+                await createVote({
+                    id: post.postId,
+                    voteAction: cancelAction,
+                    votedNonce: post.votedNonce,
+                    votedEpoch: post.votedEpoch,
+                })
+            }
+            await createVote({
+                id: post.postId,
+                voteAction: voteType,
+                votedNonce: null,
+                votedEpoch: null,
+            })
+            if (voteType === VoteAction.UPVOTE) {
+                post.upCount += 1
+            } else if (voteType === VoteAction.DOWNVOTE) {
+                post.downCount += 1
+            }
```

Comment:
It seems it is not necessary to update `post.upCount` and `post.downCount` in the `handleVote` function.

Reply:
fixed